File: Data Structures & Algorithms/anagram-groups/submission-0.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> res = new ArrayList<>();
        List<String> characters = new ArrayList<>();
        for (int i = 0; i < strs.length; i++) {
            int[] list = new int[26];
            for (int j = 0; j < strs[i].length(); j++) {
                list[strs[i].charAt(j) - 'a']++;
            }
            characters.add(Arrays.toString(list));

        }

        HashMap<String, Integer> anaFound = new HashMap<>();
        int b = 0;
        for (int a = 0; a < strs.length; a++) {
            if (anaFound.get(characters.get(a)) != null) {
                (res.get(anaFound.get(characters.get(a)))).add(strs[a]);
            } else {
                List<String> substrings = new ArrayList<>();
                substrings.add(strs[a]);
                anaFound.put(characters.get(a), b);
                b++;
                res.add(substrings);
            }
        }

        return res;
    }
}

File: Data Structures & Algorithms/pow-x-n/submission-0.cpp
class Solution {
public:
    double myPow(double x, int n) {
        if (n == 1 || x == 0) {
            return x;
        }
        if (n == 0) {
            return 1;
        }
        if (n > 0) {
            return x * myPow(x, n - 1);
        } else {
            return myPow(x, n + 1) / x;
        }
    }
};
